modules/core/04-channel/keeper/timeout.go

```diff
@@ -153,8 +153,14 @@ func (k Keeper) TimeoutExecuted(
 
 	k.deletePacketCommitment(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
 
+	if channel.FlushStatus == types.FLUSHING && !k.hasInflightPackets(ctx, packet.GetSourcePort(), packet.GetSourceChannel()) {
```

Comment:
P.s, these could also be:

```go
	switch channel.Ordering {
	case types.ORDERED:
		channel.State = types.CLOSED
		channel.FlushStatus = types.NOTINFLUSH
	case types.UNORDERED:
		if channel.FlushStatus == types.FLUSHING && !k.hasInflightPackets(ctx, packet.GetSourcePort(), packet.GetSourceChannel()) {
			channel.FlushStatus = types.FLUSHCOMPLETE
		}
	}
	k.SetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), channel)
```

and, considering `ORDERED_ALLOW_TIMEOUT`, this could be amended in the future to possible be `case types.UNORDERED, types.ORDERED_ALLOW_TIMEOUT` since it looks like it would require similar handling.

Happy to leave as is but dropping this sionce I noticed it.

Reply:
i think the switch statement is nice for readability! 

but i guess we probably need to check `if channel.FlushStatus == types.FLUSHING && !k.hasInflightPackets(ctx, packet.GetSourcePort(), packet.GetSourceChannel())` in both cases rather than just in the `UNORDERED` case, and then basically if it's ORDERED set to CLOSED/NOTINFLUSH, and if it's UNORDERED set to FLUSHCOMPLETE. but i'm not super opinionated on which style we go with.